src/com/owncloud/android/utils/FileStorageUtils.java

```diff
@@ -211,86 +217,163 @@ public static RemoteFile fillRemoteFile(OCFile ocFile){
         file.setPrivateLink(ocFile.getPrivateLink());
         return file;
     }
-    
+
     /**
-     * Sorts all filenames, regarding last user decision 
+     * Sort all OCFile Objects
+     * Used for sorting files that have already been uploaded
+     *
+     * @param files
      */
-    public static Vector<OCFile> sortFolder(Vector<OCFile> files){
-        switch (mSortOrder){
-        case SORT_NAME:
-            files = FileStorageUtils.sortByName(files);
-            break;
-        case SORT_DATE:
-            files = FileStorageUtils.sortByDate(files);
-            break;
-        case SORT_SIZE:
-            files = FileStorageUtils.sortBySize(files);
-            break;
+    public static Vector<OCFile> sortFolderFileDisp(Vector<OCFile> files) {
```

Comment:
I think that instead of including methods to work with files, maybe is better to keep using OCFile everywhere and create OCFile from files by using https://github.com/dlech/owncloud-android/blob/master/src/com/owncloud/android/datamodel/OCFile.java#L69, what do you think?

Reply:
@davigonz I have already tried this and this already has a problem which is:-
When we convert File object to OCFile object the name of the file can be accessed with `getFileName()` method but the other attributes like last modified time and file size cannot be obtained using the corresponding methods `getModificationTimestamp()` and `getFileLength() `of OCFile object hence affecting the sortBySize and sortByDate functionality. Here is a small experiment:- 
This is the LocalFileListAdapter.java where I have used the OCFile constructor to create a OCFile object of the first File object present in the mFiles Array.
<img width="489" alt="screenshot 2018-10-09 at 6 59 33 pm" src="https://user-images.githubusercontent.com/22047421/46673036-80e9cf00-cbf6-11e8-8f1d-3dafb379f180.png">
Here are the logs.
<img width="1369" alt="screenshot 2018-10-09 at 6 59 40 pm" src="https://user-images.githubusercontent.com/22047421/46673037-80e9cf00-cbf6-11e8-87dd-78008d7f5d83.png">
As you an see when being converted to OCFile the methods `getModificationTimestamp()` and `getFileLength()` giving output as 0 which is not the actual case. Hence the problem.

Reply:
@davigonz I have made rest of the changes other than this.

Reply:
@davigonz Yes this could work I will work on this implementation.